packages/dapp/components/staking/staking-shares-explorer.tsx

```diff
@@ -44,91 +44,95 @@ type Actions = {
-
-    const shares: ShareData[] = [];
-    await Promise.all(
+    const { stakingFacet, chefFacet, stakingShare: stakingToken, curveMetaPoolDollarTriPoolLp } = await protocolContracts;
+    if (stakingFacet) {
+      console.time("StakingShareExplorerContainer contract loading");
+      const currentBlock = await web3Provider.getBlockNumber();
       // cspell: disable-next-line
-      bondingShareIds.map(async (id: BigNumber) => {
+      const blockCountInAWeek = +(await stakingFacet.blockCountInAWeek()).toString();
+      const totalShares = await chefFacet?.totalShares();
+      // cspell: disable-next-line
+      const stakingShareIds = await stakingToken?.holderTokens(walletAddress);
+
+      const walletLpBalance = await curveMetaPoolDollarTriPoolLp?.balanceOf(walletAddress);
+
+      const shares: ShareData[] = [];
+      await Promise.all(
         // cspell: disable-next-line
-        const [ugov, bond, bondingShareInfo, tokenBalance] = await Promise.all([
-          // cspell: disable-next-line
-          masterChef.pendingGovernance(id),
+        stakingShareIds.map(async (id: BigNumber) => {
           // cspell: disable-next-line
-          bondingToken.getStake(id),
-          masterChef.getStakingShareInfo(id),
-          // cspell: disable-next-line
-          bondingToken.balanceOf(walletAddress, id),
-        ]);
+          const [governanceToken, stake, stakingShareInfo, tokenBalance] = await Promise.all([
+            // cspell: disable-next-line
+            chefFacet?.pendingGovernance(id),
+            // cspell: disable-next-line
+            stakingToken?.getStake(id),
+            chefFacet?.getStakingShareInfo(id),
+            // cspell: disable-next-line
+            stakingToken?.balanceOf(walletAddress, id),
+          ]);
 
-        const endBlock = +bond.endBlock.toString();
-        const blocksLeft = endBlock - currentBlock;
-        const weeksLeft = Math.round((blocksLeft / blockCountInAWeek) * 100) / 100;
+          const endBlock = +stake.endBlock.toString();
+          const blocksLeft = endBlock - currentBlock;
+          const weeksLeft = Math.round((blocksLeft / blockCountInAWeek) * 100) / 100;
 
-        // If this is 0 it means the share ERC1155 token was transferred to another account
-        if (+tokenBalance.toString() > 0) {
-          // cspell: disable-next-line
-          shares.push({ id: +id.toString(), ugov, bond, sharesBalance: bondingShareInfo[0], weeksLeft });
-        }
-      })
-    );
-
-    const sortedShares = shares.sort((a, b) => a.id - b.id);
-
-    console.timeEnd("BondingShareExplorerContainer contract loading");
-    setModel((model) => ({
-      processing: model ? model.processing.filter((id) => id !== processedShareId) : [],
-      shares: sortedShares,
-      totalShares,
-      walletLpBalance,
-    }));
+          // If this is 0 it means the share ERC1155 token was transferred to another account
+          if (+tokenBalance.toString() > 0) {
```

Comment:
why we should force the `tokenBalance` to be positive, couldn't you handle an error?